src/nvim/ex_getln.c

```diff
@@ -1669,7 +1672,16 @@ getcmdline (
     int indent               // indent for inside conditionals
 )
 {
-  return command_line_enter(firstc, count, indent);
+  if (ui_is_external(kUICmdline)) {
+    Array args = ARRAY_DICT_INIT;
+    ui_event("cmdline_enter", args);
```

Comment:
Separation of concerns. We do want richer mode change events, but that's relevant not only for ui, and it doesn't belong in `ext_cmdline`. These events should just tell the ui the current cmdline state to draw.